src/test/java/com/devonfw/sample/archunit/SecurityTest.java

```diff
@@ -31,4 +57,28 @@ public class SecurityTest {
                     .orShould().beAnnotatedWith(DenyAll.class)
                     .because("All Use-Case implementation methods must be annotated with a security " +
                             "constraint from javax.annotation.security");
+
+    /**
+     * Checks if these methods are being used
+     * Query createQuery(String qlString)
+     * <T> TypedQuery<T> createQuery(String qlString, Class<T> resultClass)
+     * Query createNativeQuery(String sqlString)
+     * Query createNativeQuery(String sqlString, Class resultClass)
+     * Query createNativeQuery(String sqlString, String resultSetMapping)
+     */
+    @ArchTest
+    private static final ArchRule shouldnTUseCreateQuery = noClasses().should().callMethodWhere(new DescribedPredicate<JavaMethodCall>("test whether CreateQuery or CreateNativQuery is used") {
+            @Override
+            public boolean test(JavaMethodCall javaMethod) {
+                        if(javaMethod.getName().equals("createQuery")){
+                                System.out.println(javaMethod.getTarget().getParameterTypes());
+                                Matcher matcherCQ = PARAMETER_PATERN_CQ.matcher(javaMethod.getTarget().getParameterTypes().toString());
```

Comment:
We should not rely on external `toString()` representations. Those are typically unstable and likely to change.
Simply further decompose the `ParameterTypes` to implement the check in a more reliable way.

Reply:
the ParameterTypes return a List of JavaClasses, with the Name of that class being the Type e.g. [JavaClass{name=java.lang.String}]  so I tried changing it to use the getName() method to compare instead of toString. Hope this is a more reliable way.